DOC-2392: Re-write of `editor-plugin-version.adoc` page. (#3292)
tinymce/tinymce-docs: modules/ROOT/pages/editor-plugin-version.adoc

> = Specify editor & plugins

Reviewer: This doesn't read right to me. Should the title be `Specify editor version & plugins`?

Author: agree

Author: ```suggestion
= Specify editor version & plugins
```

> :description_short: Specifying editor version and plugins for Tiny Cloud deployments.
> :description: Instructions on setting the TinyMCE editor version and plugins for Tiny Cloud deployments.

Reviewer: ```suggestion
:description_short: Specifying editor version and plugins for {cloudname} deployments.
:description: Instructions on setting the {productname} editor version and plugins for {cloudname} deployments.
```

:keywords: tinymce, cloud, script, textarea, apiKey, hybrid

[[specifying-the-tinymce-editor-version-deployed-from-cloud]]
== Specifying the {productname} Editor Version from the Cloud

To set the {productname} version when deploying from {cloudname}, use the following URL:

[source,html,subs="attributes+"]
----
<script src="{cdnurl}" referrerpolicy="origin"></script>
----

The example above shows the default way to load {productname} {productmajorversion} from {cloudname}. This URL ensures the most recent and verified version of {productname} is used.

[TIP]
Replace 'no-api-key' with your own API key.

For more details, check out the xref:editor-and-features.adoc[{productname} editor via {cloudname}].

[[specifying-the-cloud-deployment-of-plugins]]
== Specifying the Cloud Deployment of Plugins

> When deploying {productname}, integrators have the choice to **self-host** premium plugins or load them from {companyname} Cloud (a.k.a. hybrid mode). Depending on your requirements, you can use a combination of both approaches.

Reviewer 2: Hopefully this will avoid mistakes like attempting to load both the editor and plugins from the cloud (which isn't necessary because the editor includes the plugins)
```suggestion
When self-hosting {productname}, integrators have the choice to host premium plugins or load them from {companyname} Cloud (a.k.a. hybrid mode). Depending on your requirements, you can use a combination of both approaches.
```

Author: Just to confirm, do we refer to it as hybrid, or do we use any other term.

Reviewer 2: it's hybrid.

include::partial$misc/admon-script-tag-placement.adoc[]

{companyname} offers the following configuration scripts to assist:

=== `plugin.min.js` Standalone with No Exclusions.

> The `plugins.min.js` script loads every plugin the API key is entitled to from the CDN.

Reviewer 2: ```suggestion
The `plugins.min.js` script loads every premium plugin the API key is entitled to from the CDN.
```

[source,html,subs="attributes+"]
----
<script src="https://cdn.tiny.cloud/1/api-key/tinymce/7/plugins.min.js" referrerpolicy="origin"></script>
----

=== `plugins.min.js` with Exclusions for Specific Plugins.

To exclude specific premium plugins from `plugins.min.js` because you are self-hosting them, add the excluded plugins to the script query parameters as shown below:

[source,html,subs="attributes+"]
----
<script src="https://cdn.tiny.cloud/1/api-key/tinymce/7/plugins.min.js?mentions=sdk&powerpaste=sdk" referrerpolicy="origin"></script>
----

[NOTE]

> When excluding specific plugins from the CDN, ensure that the excluded plugins' names are appended with `=sdk`, such as `mentions=sdk`. Note that when {companyname} Cloud releases a new plugin, you will need to update your script tag to accommodate it.

Reviewer 2: This is in a `note` section, I don't think we need nested notes 😂 
```suggestion
Ensure that the excluded plugins' names are appended with `=sdk`, such as `mentions=sdk`.

When {companyname} Cloud releases a new premium plugin, if you wish to self-host it this script tag will need to be updated to exclude it.
```

----
<script src="https://cdn.tiny.cloud/1/api-key/tinymce/7/cloud-plugins.min.js?mentions&powerpaste&advcode" referrerpolicy="origin"></script>
----

[NOTE]
This approach is intended for edge cases where you self-host most premium plugins on your own servers and only need one or two from {companyname} Cloud.

'''

=== Selecting Specific Editor Versions

[IMPORTANT]
====
All {cloudname} channels are based on the {enterpriseversion} version.

* For information on the latest version of the {cloudname} `{productmajorversion}` release channel, see: xref:release-notes.adoc[{productname} Release Notes].
* For a list of changes that *may* be present in the {cloudname} testing channel, see: xref:changelog.adoc[{productname} Changelog].
====

[#{productmajorversion}-{productmajorversion}-testing-and-{productmajorversion}-dev-release-channels]
=== Difference between using {productname} {productmajorversion}, {productmajorversion}-Testing, and {productmajorversion}-Dev release channels

Choose from the `{productmajorversion}`, `{productmajorversion}-testing`, or `{productmajorversion}-dev` release channels to load the latest version of {productname} from {cloudname}.

> These channels are updated automatically and provide the latest {productname} version based on the follow criteria:

Reviewer 2: ```suggestion
These channels are updated automatically and provide the latest {productname} version based on the following criteria:
```

[#example-using-the-{productmajorversion}-release-channel]
===== Example: using the `{productmajorversion}` release channel

[source,html,subs="attributes+"]
----
<script src="{cdnurl}" referrerpolicy="origin"></script>
----

[#{productmajorversion}-testing-release-channel]
==== {productmajorversion}-Testing Release Channel

This channel deploys the current *release candidate* for the `{productmajorversion}` channel. The {productname} release candidate is undergoing quality assurance. The current version of {productname} available through the `{productmajorversion}-testing` channel can be found on the https://cdn.tiny.cloud/1/no-api-key/tinymce/{productmajorversion}-testing/version.txt[{cloudname} {productname} {productmajorversion}-testing version page].

[#example-using-the-{productmajorversion}-testing-release-channel]
===== Example: Using the `{productmajorversion}-testing` Release Channel

[source,html,subs="attributes+"]
----
<script src="https://cdn.tiny.cloud/1/no-api-key/tinymce/{productmajorversion}-testing/tinymce.min.js" referrerpolicy="origin"></script>
----

[#{productmajorversion}-dev-release-channel]
==== {productmajorversion}-Dev Release Channel

> This channel deploys nightly builds of {productname}, which includes unreleased changes documented in the https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/CHANGELOG.md[{productname} changelog]. The current version of {productname} available through the `{productmajorversion}-dev` channel can be found on the https://cdn.tiny.cloud/1/no-api-key/tinymce/{productmajorversion}-dev/version.txt[{cloudname} {productname} {productmajorversion}-dev version page].

Reviewer: The current dev versions is 7.2, but the changelog link mentioned here doesn't include 7.2 changes. Is this statement still correct?

Author: @ltrouton Can you advise if the above is still accurate based on changes leading up to `7.0`

Reviewer 2: the in-repo changelog is no longer updated before the next release. I don't know if we publish the "unreleased changelog" anywhere now.

Maybe just change `documented in the changelog` to `from the tinymce repository`. Link to the repo (noting we use branch `main` now not `develop`).

Author: ```suggestion
This channel deploys nightly builds of {productname}, which includes **unreleased changes** from the link:https://github.com/tinymce/tinymce/blob/main/modules/tinymce/CHANGELOG.md[{productname} changelog^]. 

The current version of {productname} is available on the `{productmajorversion}-dev` channel can be found on the link:https://cdn.tiny.cloud/1/no-api-key/tinymce/{productmajorversion}-dev/version.txt[{cloudname} {productname} {productmajorversion}-dev version page^].

[NOTE]
{productname} now uses `main` branch for nightly builds instead of `develop` as of {productname} `7.0.0`.
```

Reviewer 2: That still links to the changelog. How about

```
This channel deploys nightly builds of {productname}, which includes **unreleased changes** from the link:https://github.com/tinymce/tinymce/tree/main[{productname} repository^]. 
```

* `plugins.min.js`: This script configures {productname} to load all premium plugins from {companyname} Cloud, **defaulting** to the `latest` version in the selected cloud channel. You can specify different plugin versions if necessary, but it's generally best to use the latest versions for optimal performance and security.
* `cloud-plugins.min.js`: This script allows loading of specific premium plugins from {companyname} Cloud, while self-hosting the rest. This approach is intended for edge cases, such as when you have custom plugins on your own servers and only need a few from {companyname} Cloud.

=== Which Script Should You Use?

==== Using `plugins.min.js`

In most cases, using `plugins.min.js` is the correct choice. It provides the most flexibility and simplicity by ensuring all plugins are loaded from {companyname} Cloud. If you need to specify individual plugin versions, you can do so via query parameters.

.Example

> [source,html]

Author: I believe we covered this in the discussion on slack, closing this for re-review of new `commit`.

----

> <script src="https://cdn.tiny.cloud/1/no-api-key/tinymce/7/plugins.min.js?mentions=3.0&powerpaste=6.0&advcode=sdk" referrerpolicy="origin"></script>

Reviewer 3: What does `advcode=sdk` mean?

----

In this example, {productname} will try to load all premium plugins from {companyname} Cloud, including specified versions for the "mentions" and "powerpaste" plugins. The disadvantage of using `plugins.min.js` is that if you want to load only one plugin from {companyname} Cloud while the rest are self-hosted, you have to add all other plugins as query parameters with their versions specified as "sdk." When {companyname} Cloud releases a new plugin, you'll need to update the list.

include::partial$misc/admon-script-tag-placement.adoc[]

==== Using `cloud-plugins.min.js`

{companyname} recommends only using `cloud-plugins.min.js` if you have a specific reason to self-host some premium plugins while loading others from Tiny Cloud. This approach is more complex and requires you to specify which plugins to load from {companyname} Cloud.

.Example
[source,html]
----

> <script src="https://cdn.tiny.cloud/1/no-api-key/tinymce/7/cloud-plugins.min.js?mentions=3.0&powerpaste=6.0&advcode" referrerpolicy="origin"></script>

Reviewer 3: Going forward from 7, the versions of plugins should not matter anymore since they are the same as the tinymce version? In this case, what is the version of advcode if the version is not specified?

Reviewer 2: The versions still technically matter, but it's never been clear to me why customers might ever want this. They only do because it's the first example in our docs.

Likewise it's not clear why self-hosted plugins are specified in the `cloud-plugins` url. We probably need to look at what the server actually returns before we proceed much further with this effort.

Author: Removed any mention of plugin-versions in last commit, resolving this conversation as its no longer relevant in new approach.
